docs/commands.rst

```diff
@@ -0,0 +1,213 @@
+  initializes a :py:class:`FindTransactionCommand` object with the adapter of the
+  API instance it belongs to.
+
+- Then calls this command with the keyword arguments it was provided with.
+
+- The command prepares the request by applying a ``RequestFilter`` on the
+  payload. The command specific ``RequestFilter`` validates that the payload
+  has correct types, in some cases it is even able to convert the payload to
+  the required type and format.
+
+- Command execution injects the name of the API command (see `IRI API Reference`_
+  for command names) in the request and sends it to the adapter.
+
+- The adapter communicates with the node and returns its response.
+
+- The response is prepared by going through a command specific ``ResponseFilter``.
+
+- The response is returned to the high level API method as a ``dict``, ready
+  to be returned to the main application.
+
+.. note::
+    A command object can only be called once without resetting it. When you
+    use the high level API methods, you don't need to worry about resetting
+    commands as each call to an API method will initialize a new command object.
+
+Filters
+-------
+
+If you take a look on the actual implementation of :py:class:`FindTransactionsCommand`,
+you notice that you have to define your own request- and response filter classes.
+
+Filters in PyOTA are based on the `Filters library`_. Read more about how
+they work on the `filters documentation site`_.
+
+In short, you can create filter chains through which the filtered value passes,
+and generates errors if something failed validation. Filter chains are specified
+in the custom filter class's :py:meth:`__init__` function. If you also want to
+modify the filtered value before returning it, override the :py:meth:`_apply`
+method of its base class. Read more about how to `create custom filters`_.
+
+PyOTA offers you some custom filters for PyOTA specific types:
+
+**Trytes**
+~~~~~~~~~~
+.. autoclass:: iota.filters.Trytes
+
+**StringifiedTrytesArray**
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. automethod:: iota.filters.StringifiedTrytesArray
+
+**AddressNoChecksum**
+~~~~~~~~~~~~~~~~~~~~~
+.. autoclass:: iota.filters.AddressNoChecksum
+
+**GeneratedAddress**
+~~~~~~~~~~~~~~~~~~~~
+.. autoclass:: iota.filters.GeneratedAddress
+
+**NodeUri**
+~~~~~~~~~~~
+.. autoclass:: iota.filters.NodeUri
+    :members: SCHEMES
+
+**SecurityLevel**
+~~~~~~~~~~~~~~~~~~~~
+.. automethod:: iota.filters.SecurityLevel
+
+.. important::
+    The general rule in PyOTA is that all requests going to a node are validated
+    , but only responses that contain transaction/bundle trytes or hashes are
+    checked.
+
+    Also note, that for extended commands, ``RequestFilter`` is usually
+    implemented with just a "pass" statement. The reason being that these
+    commands do not directly send their payload to a node, but rather call
+    core commands that do have their ``RequestFilter`` implemented.
+    More about this topic in the next section.
```

Comment:
❔ I think this applies to `ResponseFilter` rather than `RequestFilter`.

Reply:
yes, you are right.